### DISCUSSION TITLE
How to load components dynamically based on the name from API

### DISCUSSION THREAD

**Discussion by adityar15** [nuxt/nuxt#23015] [General]
I have 100s of Vue components in a Nuxt project. I am using headless CMS and the component name comes from the CMS API. Now the problem here is that if I want all of these components to be available on demand, I have to register them globally which might not be efficient. 

I tried defineAsyncComponent from vue but it failed to find the component chunk.js

```
const Component = defineAsyncComponent(() => import(`./${props.componentName}`))
```

The resovleComponent does not work as it needs a literal string and not a variable.
```
const Component = resolveComponentt(`./${props.componentName}`)
```

Is there any other way to dynamically load components based on the name from db or API?

**Comment by rrubio** [comment]
+1

**Comment by milelazar** [comment]
Do you have 100s of components that must be nuxt pages or this includes components that needs to be loaded within pages? I have the same scenario but there is no need for all components to be registered as global so it works fine so far.

I ended up with one [...slug] catch all page. Page has dynamic component that loads the right global component/page/template based on routes and API data. All other subcomponents within page are just regular components. No need for them to be global.

Here is an example: https://stackblitz.com/edit/nuxt-starter-ezoqnu?file=components%2Fglobal%2Fcms%2FTemplate3.vue
Template conditions are hardcoded just to simplify the example but works fine with api data.

**Comment by adityar15** [comment]
I am using a headless CMS and the component's name comes from there. Ideally, it would be beneficial to have many components registered beforehand.

Currently, I am using a workaround where the components that are global are registered in a folder and then referenced using that folder name. This way the dashboard components are not global while the marketing site components are. I referred from Nuxt documentation's global component section
